pkg/operator/render.go

```diff
@@ -68,30 +69,16 @@ func createDiscoveredControllerConfigSpec(infra *configv1.Infrastructure, networ
 	}
 
 	platform := "none"
-	//nolint:staticcheck
-	switch infra.Status.Platform {
-	case configv1.AWSPlatformType:
-		platform = "aws"
-	case configv1.AzurePlatformType:
-		platform = "azure"
-	case configv1.BareMetalPlatformType:
-		platform = "baremetal"
-	case configv1.GCPPlatformType:
-		platform = "gcp"
-	case configv1.OpenStackPlatformType:
-		platform = "openstack"
-	case configv1.LibvirtPlatformType:
-		platform = "libvirt"
-	case configv1.VSpherePlatformType:
-		platform = "vsphere"
+	infraPlatformString := string(infra.Status.PlatformStatus.Type)
```

Comment:
This will most likely fail 4.1 to 4.2 upgrades

Reply:
You should be using the .platformStatus only when set and fall back to old .platformType when not

Reply:
Taken care of. Thanks!